esiclient/tests/functional/v1/test_node.py

```diff
@@ -0,0 +1,111 @@
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import ddt
+from tempest.lib import exceptions
+
+from esiclient.tests.functional import base
+from esiclient.tests.functional import utils
+
+
+@ddt.ddt
+class NodePowerTests(base.ESIBaseTestClass):
+    """Functional tests for esi node power commands."""
+
+    @classmethod
+    def setUpClass(cls):
+        super(NodePowerTests, cls).setUpClass()
+        cls._init_dummy_project(cls, 'owner_proj', 'owner')
+        cls._init_dummy_project(cls, 'owner_proj_2', 'owner')
+        cls._init_dummy_project(cls, 'lessee_proj', 'lessee')
+        cls._init_dummy_project(cls, 'lessee_proj_2', 'lessee')
+
+    def setUp(self):
+        super(NodePowerTests, self).setUp()
+        self.clients = NodePowerTests.clients
+        self.users = NodePowerTests.users
+        self.projects = NodePowerTests.projects
+
+    def test_owner_node_power_on_off(self):
+
+        node = utils.node_create(self.clients['admin'],
+                                 name="owned")
+        self.addCleanup(utils.node_delete,
+                        self.clients['admin'],
+                        node['name'])
+
+        utils.node_set(self.clients['admin'],
+                       identifier=node['name'],
+                       param='owner',
+                       value=self.projects['owner_proj']['id'])
+
+        utils.node_power_on(self.clients['owner_proj-owner'], node['name'])
+        utils.node_power_off(self.clients['owner_proj-owner'], node['name'])
+
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.node_power_on,
+                          self.clients['owner_proj_2-owner'],
+                          node['name'])
+
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.node_power_off,
+                          self.clients['owner_proj_2-owner'],
+                          node['name'])
+
+    def test_lessee_node_power_on_off(self):
+
+        node = utils.node_create(self.clients['admin'],
+                                 name="owned")
+        self.addCleanup(utils.node_delete,
+                        self.clients['admin'],
+                        node['name'])
+
+        utils.node_set(self.clients['admin'],
+                       identifier=node['name'],
+                       param='owner',
+                       value=self.projects['owner_proj']['id'])
+
+        utils.node_set(self.clients['admin'],
+                       identifier=node['name'],
+                       param='lessee',
+                       value=self.projects['lessee_proj']['id'])
+
+        utils.node_power_on(self.clients['lessee_proj-lessee'], node['name'])
+        utils.node_power_off(self.clients['lessee_proj-lessee'], node['name'])
+
+        self.assertRaises(exceptions.CommandFailed,
```

Comment:
This also should probably be split into a separate negative test case.

Reply:
agreed